Wrong `no-unused-style` warning with conditional styles

Sample code to reproduce:

```js
import React, { Component, PropTypes } from 'react';
import ReactNative from 'react-native';

const {
	StyleSheet,
	View
} = ReactNative;

const styles = StyleSheet.create({
	style1: {
		color: 'red',
	},
	style2: {
		color: 'blue',
	}
});

export default class MyComponent extends Component {
	static propTypes = {
		isDanger: PropTypes.bool
	};

	render() {
		return <View style={this.props.isDanger ? styles.style1 : styles.style2} />;
	}
}
```

ESLint output:

```sh
/path/to/test.js
  10:2  warning  Unused style detected: styles.style1  react-native/no-unused-styles
  13:2  warning  Unused style detected: styles.style2  react-native/no-unused-styles

✖ 2 problems (0 errors, 2 warnings)
```

ESLint config:

```
    "react-native/no-unused-styles": 1,
    "react-native/split-platform-components": 1,
```

--- a/.eslintrc
+++ b/.eslintrc
@@ -5,5 +5,6 @@
     "func-names": 0,
     "object-shorthand": 0,
     "consistent-return": 0,
+    "prefer-template": 0
   }
 }

--- a/.eslintrc
+++ b/.eslintrc
@@ -1,0 +1,9 @@
+{
+  "extends": "airbnb",
+  rules: {
+    "strict": [0, "global"],
+    "func-names": 0,
+    "object-shorthand": 0,
+    "consistent-return": 0,
+  }
+}

--- a/lib/rules/no-unused-styles.js
+++ b/lib/rules/no-unused-styles.js
@@ -3,174 +3,25 @@
  * @author Tom Hastjarjanto
  */
 'use strict';
-var Components = require('../util/Components');
-var StyleSheets = require('../util/stylesheet').StyleSheets;
+const Components = require('../util/Components');
+const styleSheet = require('../util/stylesheet');
+const StyleSheets = styleSheet.StyleSheets;
+const astHelpers = styleSheet.astHelpers;
 
-module.exports =  Components.detect(function(context, components, utils) {
-  var sourceCode = context.getSourceCode();
-  var styleSheets = new StyleSheets();
-  var styleReferences = new Set();
+module.exports = Components.detect((context, components) => {
+  const styleSheets = new StyleSheets();
+  const styleReferences = new Set();
 
-  function containsStyleSheetObject(node) {
-    return Boolean(
-      node &&
-      node.init &&
-      node.init.callee &&
-      node.init.callee.object &&
-      node.init.callee.object.name === 'StyleSheet'
-    );
-  }
-
-  function containsCreateCall(node) {
-    return Boolean(
-      node &&
-      node.init &&
-      node.init.callee &&
-      node.init.callee.property &&
-      node.init.callee.property.name === 'create'
-    );
-  }
-
-  function isStyleSheetDeclaration(node) {
-    return Boolean(
-      containsStyleSheetObject(node) &&
-      containsCreateCall(node)
-    );
-  }
-
-  function getStyleSheetName(node) {
-    if (
-      node &&
-      node.id
-    ) {
-      return node.id.name;
-    }
-  }
-
-  function getStyleDeclarations(node) {
-    if (
-      node &&
-      node.init &&
-      node.init.arguments &&
-      node.init.arguments[0] &&
-      node.init.arguments[0].properties
-    ) {
-      return node
-        .init
-        .arguments[0]
-        .properties;
-    }
-
-    return [];
-  }
-
-  function isStyleAttribute(node) {
-    return Boolean(
-      node.type === 'JSXAttribute' &&
-      node.name &&
-      node.name.name &&
-      node.name.name.toLowerCase().includes('style')
-    );
-  }
-
-  function collectStyleReferences(node) {
-    if (hasArrayOfStyleReferences(node)) {
-      var styleReferenceContainers = node
-        .expression
-        .elements;
-
-      return collectStyleReferencesFromContainers(
-        styleReferenceContainers
-      );
-    } else {
-      return [getStyleReferenceFromExpression(node.expression)];
-    }
-  }
-
-  function collectStyleReferencesFromContainers(nodes) {
-    var styleReferences = [];
-    nodes.forEach(function(node) {
-      styleReferences = styleReferences.concat(getStyleReferenceFromNode(node));
-    });
-
-    return styleReferences;
-  }
-
-  function getStyleReferenceFromNode(node) {
-    switch (node.type) {
-      case 'MemberExpression':
-        var styleReference = getStyleReferenceFromExpression(node);
-        return [styleReference];
-        break;
-      case 'LogicalExpression':
-        var leftStyleReferences = getStyleReferenceFromNode(node.left);
-        var rightStyleReferences = getStyleReferenceFromNode(node.right);
-        return [].concat(leftStyleReferences).concat(rightStyleReferences);
-        break;
-      case 'ConditionalExpression':
-        var leftStyleReferences = getStyleReferenceFromNode(node.consequent);
-        var rightStyleReferences = getStyleReferenceFromNode(node.alternate);
-        return [].concat(leftStyleReferences).concat(rightStyleReferences);
-        return [];
-        break;
-      default:
-        return [];
-    }
-  }
-
-  function hasArrayOfStyleReferences(node) {
-    return Boolean(
-      node.type === 'JSXExpressionContainer' &&
-      node.expression &&
-      node.expression.type === 'ArrayExpression'
-    );
-  }
-
-  function getStyleReferenceFromExpression(node) {
-    var result = [];
-    var name = getObjectName(node);
-    if (name) {
-      result.push(name);
-    }
-
-    var property = getPropertyName(node);
-    if (property) {
-      result.push(property);
-    }
-
-    return result.join('.');
-  }
-
-  function getObjectName(node) {
-    if (
-      node &&
-      node.object &&
-      node.object.name
-    ) {
-      return node.object.name;
-    }
-  }
-
-  function getPropertyName(node) {
-    if (
-      node &&
-      node.property &&
-      node.property.name
-    ) {
-      return node.property.name;
-    }
-  }
-
-  function reportUnusedStyles(styleSheets) {
-    for(var key in styleSheets) {
-      if (styleSheets.hasOwnProperty(key)) {
-        var styles = styleSheets[key];
-        styles.forEach(function(node) {
-          var message = [
+  function reportUnusedStyles(unusedStyles) {
+    for (const key in unusedStyles) {
+      if (unusedStyles.hasOwnProperty(key)) {
+        const styles = unusedStyles[key];
+        styles.forEach((node) => {
+          const message = [
             'Unused style detected: ',
             key,
             '.',
-            node.key.name
+            node.key.name,
           ].join('');
 
           context.report(node, message);
@@ -180,33 +31,33 @@ module.exports =  Components.detect(function(context, components, utils) {
   }
 
   return {
-    VariableDeclarator: function(node) {
-      if (isStyleSheetDeclaration(node)) {
-        var styleSheetName = getStyleSheetName(node);
-        var styles = getStyleDeclarations(node);
+    VariableDeclarator: function (node) {
+      if (astHelpers.isStyleSheetDeclaration(node)) {
+        const styleSheetName = astHelpers.getStyleSheetName(node);
+        const styles = astHelpers.getStyleDeclarations(node);
 
         styleSheets.add(styleSheetName, styles);
       }
     },
 
-    JSXAttribute: function(node) {
-      if (isStyleAttribute(node)) {
-        var styles = collectStyleReferences(node.value);
-        styles.forEach(function(style) {
+    JSXAttribute: function (node) {
+      if (astHelpers.isStyleAttribute(node)) {
+        const styles = astHelpers.collectStyleReferences(node.value);
+        styles.forEach((style) => {
           styleReferences.add(style);
         });
       }
     },
 
-    'Program:exit': function() {
-      var list = components.list();
+    'Program:exit': function () {
+      const list = components.list();
       if (Object.keys(list).length > 0) {
-        styleReferences.forEach(function(reference){
+        styleReferences.forEach((reference) => {
           styleSheets.markAsUsed(reference);
         });
         reportUnusedStyles(styleSheets.getUnusedReferences());
       }
-    }
+    },
   };
 });
 

--- a/lib/rules/no-unused-styles.js
+++ b/lib/rules/no-unused-styles.js
@@ -4,53 +4,7 @@
  */
 'use strict';
 var Components = require('../util/Components');
-
-/**
- * StyleSheets represents the StyleSheets found in the source code.
- * @constructor
- */
-function StyleSheets() {
-  this._styleSheets = {};
-}
-
-/**
- * Add adds a StyleSheet to our StyleSheets collections.
- *
- * @param {string} styleSheetName - The name of the StyleSheet.
- * @param {object} properties - The collection of rules in the styleSheet.
- */
-StyleSheets.prototype.add = function(styleSheetName, properties) {
-  this._styleSheets[styleSheetName] = properties;
-}
-
-/**
- * MarkAsUsed marks a rule as used in our source code by removing it from the
- * specified StyleSheet rules.
- *
- * @param {string} fullyQualifiedName - The fully qualified name of the rule.
- * for example 'styles.text'
- */
-StyleSheets.prototype.markAsUsed = function(fullyQualifiedName) {
-  var nameSplit = fullyQualifiedName.split('.');
-  var styleSheetName = nameSplit[0];
-  var styleSheetProperty = nameSplit[1];
-
-  if (this._styleSheets[styleSheetName]) {
-    this._styleSheets[styleSheetName] = this
-      ._styleSheets[styleSheetName]
-      .filter(function(property) {
-        return property.key.name !== styleSheetProperty;
-      });
-  }
-}
-
-/**
-* GetUnusedReferences returns all collected StyleSheets and their
-* unmarked rules.
-*/
-StyleSheets.prototype.getUnusedReferences = function() {
-  return this._styleSheets;
-}
+var StyleSheets = require('../util/stylesheet').StyleSheets;
 
 module.exports =  Components.detect(function(context, components, utils) {
   var sourceCode = context.getSourceCode();

--- a/lib/rules/split-platform-components.js
+++ b/lib/rules/split-platform-components.js
@@ -5,56 +5,56 @@
  */
 'use strict';
 
-module.exports =  function(context) {
-  var reactComponents = [];
-  var androidMessage = 'Android components should be placed in android files';
-  var iosMessage = 'IOS components should be placed in ios files';
-  var conflictMessage = 'IOS and Android components can\'t be mixed';
+module.exports = function (context) {
+  let reactComponents = [];
+  const androidMessage = 'Android components should be placed in android files';
+  const iosMessage = 'IOS components should be placed in ios files';
+  const conflictMessage = 'IOS and Android components can\'t be mixed';
 
   function getKeyValue(node) {
-    var key = node.key || node.argument;
+    const key = node.key || node.argument;
     return key.type === 'Identifier' ? key.name : key.value;
   }
 
-  function hasNodeWithName(node, name) {
-    return node.some(function(node) {
-      var nodeName = getKeyValue(node)
+  function hasNodeWithName(nodes, name) {
+    return nodes.some((node) => {
+      const nodeName = getKeyValue(node);
       return nodeName && nodeName.includes(name);
-    })
+    });
   }
 
   function reportErrors(components, filename) {
-    var containsAndroidAndIOS = (
+    const containsAndroidAndIOS = (
       hasNodeWithName(components, 'IOS') &&
       hasNodeWithName(components, 'Android')
     );
 
-    for(var i = 0; i < components.length; i++) {
-      var node = components[i];
-      var propName = getKeyValue(node);
+    for (let i = 0; i < components.length; i++) {
+      const node = components[i];
+      const propName = getKeyValue(node);
 
       if (propName.includes('IOS') && !filename.endsWith('.ios.js')) {
-        context.report(node, containsAndroidAndIOS ? conflictMessage : iosMessage)
+        context.report(node, containsAndroidAndIOS ? conflictMessage : iosMessage);
       }
 
       if (propName.includes('Android') && !filename.endsWith('.android.js')) {
-        context.report(node, containsAndroidAndIOS ? conflictMessage : androidMessage)
+        context.report(node, containsAndroidAndIOS ? conflictMessage : androidMessage);
       }
     }
   }
 
   return {
-    VariableDeclarator: function(node) {
-      var destructuring = node.init && node.id && node.id.type === 'ObjectPattern';
-      var statelessDestructuring = destructuring && node.init.name === 'React';
+    VariableDeclarator: function (node) {
+      const destructuring = node.init && node.id && node.id.type === 'ObjectPattern';
+      const statelessDestructuring = destructuring && node.init.name === 'React';
       if (destructuring && statelessDestructuring) {
         reactComponents = reactComponents.concat(node.id.properties);
       }
     },
-    'Program:exit': function() {
-      var filename = context.getFilename();
+    'Program:exit': function () {
+      const filename = context.getFilename();
       reportErrors(reactComponents, filename);
-    }
+    },
   };
 };
 

--- a/lib/util/Components.js
+++ b/lib/util/Components.js
@@ -4,7 +4,7 @@
  */
 'use strict';
 
-var util = require('util');
+const util = require('util');
 
 /**
  * Components
@@ -12,7 +12,7 @@ var util = require('util');
  */
 function Components() {
   this._list = {};
-  this._getId = function(node) {
+  this._getId = function (node) {
     return node && node.range.join(':');
   };
 }
@@ -23,8 +23,8 @@ function Components() {
  * @param {ASTNode} node The AST node being added.
  * @param {Number} confidence Confidence in the component detection (0=banned, 1=maybe, 2=yes)
  */
-Components.prototype.add = function(node, confidence) {
-  var id = this._getId(node);
+Components.prototype.add = function (node, confidence) {
+  const id = this._getId(node);
   if (this._list[id]) {
     if (confidence === 0 || this._list[id].confidence === 0) {
       this._list[id].confidence = 0;
@@ -35,7 +35,7 @@ Components.prototype.add = function(node, confidence) {
   }
   this._list[id] = {
     node: node,
-    confidence: confidence
+    confidence: confidence,
   };
 };
 
@@ -45,8 +45,8 @@ Components.prototype.add = function(node, confidence) {
  * @param {ASTNode} node The AST node being searched.
  * @returns {Object} Component object, undefined if the component is not found
  */
-Components.prototype.get = function(node) {
-  var id = this._getId(node);
+Components.prototype.get = function (node) {
+  const id = this._getId(node);
   return this._list[id];
 };
 
@@ -56,14 +56,15 @@ Components.prototype.get = function(node) {
  * @param {ASTNode} node The AST node being updated.
  * @param {Object} props Additional properties to add to the component.
  */
-Components.prototype.set = function(node, props) {
-  while (node && !this._list[this._getId(node)]) {
-    node = node.parent;
+Components.prototype.set = function (node, props) {
+  let currentNode = node;
+  while (currentNode && !this._list[this._getId(currentNode)]) {
+    currentNode = node.parent;
   }
-  if (!node) {
+  if (!currentNode) {
     return;
   }
-  var id = this._getId(node);
+  const id = this._getId(currentNode);
   this._list[id] = util._extend(this._list[id], props);
 };
 
@@ -73,9 +74,9 @@ Components.prototype.set = function(node, props) {
  *
  * @returns {Object} Components list
  */
-Components.prototype.list = function() {
-  var list = {};
-  for (var i in this._list) {
+Components.prototype.list = function () {
+  const list = {};
+  for (const i in this._list) {
     if (!this._list.hasOwnProperty(i) || this._list[i].confidence < 2) {
       continue;
     }
@@ -90,9 +91,9 @@ Components.prototype.list = function() {
  *
  * @returns {Number} Components list length
  */
-Components.prototype.length = function() {
-  var length = 0;
-  for (var i in this._list) {
+Components.prototype.length = function () {
+  let length = 0;
+  for (const i in this._list) {
     if (!this._list.hasOwnProperty(i) || this._list[i].confidence < 2) {
       continue;
     }
@@ -102,12 +103,11 @@ Components.prototype.length = function() {
 };
 
 function componentRule(rule, context) {
-
-  var sourceCode = context.getSourceCode();
-  var components = new Components();
+  const sourceCode = context.getSourceCode();
+  const components = new Components();
 
   // Utilities for component detection
-  var utils = {
+  const utils = {
 
     /**
      * Check if the node is a React ES5 component
@@ -115,7 +115,7 @@ function componentRule(rule, context) {
      * @param {ASTNode} node The AST node being checked.
      * @returns {Boolean} True if the node is a React ES5 component, false if not
      */
-    isES5Component: function(node) {
+    isES5Component: function (node) {
       if (!node.parent) {
         return false;
       }
@@ -128,7 +128,7 @@ function componentRule(rule, context) {
      * @param {ASTNode} node The AST node being checked.
      * @returns {Boolean} True if the node is a React ES6 component, false if not
      */
-    isES6Component: function(node) {
+    isES6Component: function (node) {
       if (!node.superClass) {
         return false;
       }
@@ -141,8 +141,8 @@ function componentRule(rule, context) {
      * @param {ASTNode} node The AST node being checked (must be a ReturnStatement).
      * @returns {Boolean} True if the node is returning JSX, false if not
      */
-    isReturningJSX: function(node) {
-      var property;
+    isReturningJSX: function (node) {
+      let property;
       switch (node.type) {
         case 'ReturnStatement':
           property = 'argument';
@@ -154,11 +154,11 @@ function componentRule(rule, context) {
           return false;
       }
 
-      var returnsJSX =
+      const returnsJSX =
         node[property] &&
         node[property].type === 'JSXElement'
       ;
-      var returnsReactCreateElement =
+      const returnsReactCreateElement =
         node[property] &&
         node[property].callee &&
         node[property].callee.property &&
@@ -173,7 +173,7 @@ function componentRule(rule, context) {
      *
      * @returns {ASTNode} component node, null if we are not in a component
      */
-    getParentComponent: function() {
+    getParentComponent: function () {
       return (
         utils.getParentES6Component() ||
         utils.getParentES5Component() ||
@@ -186,10 +186,10 @@ function componentRule(rule, context) {
      *
      * @returns {ASTNode} component node, null if we are not in a component
      */
-    getParentES5Component: function() {
-      var scope = context.getScope();
+    getParentES5Component: function () {
+      let scope = context.getScope();
       while (scope) {
-        var node = scope.block && scope.block.parent && scope.block.parent.parent;
+        const node = scope.block && scope.block.parent && scope.block.parent.parent;
         if (node && utils.isES5Component(node)) {
           return node;
         }
@@ -203,12 +203,12 @@ function componentRule(rule, context) {
      *
      * @returns {ASTNode} component node, null if we are not in a component
      */
-    getParentES6Component: function() {
-      var scope = context.getScope();
+    getParentES6Component: function () {
+      let scope = context.getScope();
       while (scope && scope.type !== 'class') {
         scope = scope.upper;
       }
-      var node = scope && scope.block;
+      const node = scope && scope.block;
       if (!node || !utils.isES6Component(node)) {
         return null;
       }
@@ -220,13 +220,16 @@ function componentRule(rule, context) {
      *
      * @returns {ASTNode} component node, null if we are not in a component
      */
-    getParentStatelessComponent: function() {
-      var scope = context.getScope();
+    getParentStatelessComponent: function () {
+      let scope = context.getScope();
       while (scope) {
-        var node = scope.block;
-        var isFunction = /Function/.test(node.type); // Ignore non functions
-        var isNotMethod = !node.parent || node.parent.type !== 'MethodDefinition'; // Ignore classes methods
-        var isNotArgument = !node.parent || node.parent.type !== 'CallExpression'; // Ignore arguments (callback, etc.)
+        const node = scope.block;
+        // Ignore non functions
+        const isFunction = /Function/.test(node.type);
+        // Ignore classes methods
+        const isNotMethod = !node.parent || node.parent.type !== 'MethodDefinition';
+        // Ignore arguments (callback, etc.)
+        const isNotArgument = !node.parent || node.parent.type !== 'CallExpression';
         if (isFunction && isNotMethod && isNotArgument) {
           return node;
         }
@@ -241,31 +244,32 @@ function componentRule(rule, context) {
      * @param {ASTNode} node The AST node being checked (must be a MemberExpression).
      * @returns {ASTNode} component node, null if we cannot find the component
      */
-    getRelatedComponent: function(node) {
-      var i;
-      var j;
-      var k;
-      var l;
+    getRelatedComponent: function (node) {
+      let currentNode = node;
+      let i;
+      let j;
+      let k;
+      let l;
       // Get the component path
-      var componentPath = [];
-      while (node) {
-        if (node.property && node.property.type === 'Identifier') {
-          componentPath.push(node.property.name);
+      const componentPath = [];
+      while (currentNode) {
+        if (currentNode.property && currentNode.property.type === 'Identifier') {
+          componentPath.push(currentNode.property.name);
         }
-        if (node.object && node.object.type === 'Identifier') {
-          componentPath.push(node.object.name);
+        if (currentNode.object && currentNode.object.type === 'Identifier') {
+          componentPath.push(currentNode.object.name);
         }
-        node = node.object;
+        currentNode = currentNode.object;
       }
       componentPath.reverse();
 
       // Find the variable in the current scope
-      var variableName = componentPath.shift();
+      const variableName = componentPath.shift();
       if (!variableName) {
         return null;
       }
-      var variableInScope;
-      var variables = context.getScope().variables;
+      let variableInScope;
+      const variables = context.getScope().variables;
       for (i = 0, j = variables.length; i < j; i++) {
         if (variables[i].name === variableName) {
           variableInScope = variables[i];
@@ -277,10 +281,14 @@ function componentRule(rule, context) {
       }
 
       // Find the variable declaration
-      var defInScope;
-      var defs = variableInScope.defs;
+      let defInScope;
+      const defs = variableInScope.defs;
       for (i = 0, j = defs.length; i < j; i++) {
-        if (defs[i].type === 'ClassName' || defs[i].type === 'FunctionName' || defs[i].type === 'Variable') {
+        if (
+          defs[i].type === 'ClassName' ||
+          defs[i].type === 'FunctionName' ||
+          defs[i].type === 'Variable'
+        ) {
           defInScope = defs[i];
           break;
         }
@@ -288,72 +296,72 @@ function componentRule(rule, context) {
       if (!defInScope) {
         return null;
       }
-      node = defInScope.node.init || defInScope.node;
+      currentNode = defInScope.node.init || defInScope.node;
 
       // Traverse the node properties to the component declaration
       for (i = 0, j = componentPath.length; i < j; i++) {
-        if (!node.properties) {
+        if (!currentNode.properties) {
           continue;
         }
-        for (k = 0, l = node.properties.length; k < l; k++) {
-          if (node.properties[k].key.name === componentPath[i]) {
-            node = node.properties[k];
+        for (k = 0, l = currentNode.properties.length; k < l; k++) {
+          if (currentNode.properties[k].key.name === componentPath[i]) {
+            currentNode = currentNode.properties[k];
             break;
           }
         }
-        if (!node) {
+        if (!currentNode) {
           return null;
         }
-        node = node.value;
+        currentNode = currentNode.value;
       }
 
       // Return the component
-      return components.get(node);
-    }
+      return components.get(currentNode);
+    },
   };
 
   // Component detection instructions
-  var detectionInstructions = {
-    ClassDeclaration: function(node) {
+  const detectionInstructions = {
+    ClassDeclaration: function (node) {
       if (!utils.isES6Component(node)) {
         return;
       }
       components.add(node, 2);
     },
 
-    ClassProperty: function(node) {
-      node = utils.getParentComponent();
+    ClassProperty: function () {
+      const node = utils.getParentComponent();
       if (!node) {
         return;
       }
       components.add(node, 2);
     },
 
-    ObjectExpression: function(node) {
+    ObjectExpression: function (node) {
       if (!utils.isES5Component(node)) {
         return;
       }
       components.add(node, 2);
     },
 
-    FunctionExpression: function(node) {
-      node = utils.getParentComponent();
+    FunctionExpression: function () {
+      const node = utils.getParentComponent();
       if (!node) {
         return;
       }
       components.add(node, 1);
     },
 
-    FunctionDeclaration: function(node) {
-      node = utils.getParentComponent();
+    FunctionDeclaration: function () {
+      const node = utils.getParentComponent();
       if (!node) {
         return;
       }
       components.add(node, 1);
     },
 
-    ArrowFunctionExpression: function(node) {
-      node = utils.getParentComponent();
+    ArrowFunctionExpression: function () {
+      const node = utils.getParentComponent();
       if (!node) {
         return;
       }
@@ -364,8 +372,8 @@ function componentRule(rule, context) {
       }
     },
 
-    ThisExpression: function(node) {
-      node = utils.getParentComponent();
+    ThisExpression: function () {
+      const node = utils.getParentComponent();
       if (!node || !/Function/.test(node.type)) {
         return;
       }
@@ -373,23 +381,23 @@ function componentRule(rule, context) {
       components.add(node, 0);
     },
 
-    ReturnStatement: function(node) {
+    ReturnStatement: function (node) {
       if (!utils.isReturningJSX(node)) {
         return;
       }
-      node = utils.getParentComponent();
-      if (!node) {
+      const parentNode = utils.getParentComponent();
+      if (!parentNode) {
         return;
       }
-      components.add(node, 2);
-    }
+      components.add(parentNode, 2);
+    },
   };
 
   // Update the provided rule instructions to add the component detection
-  var ruleInstructions = rule(context, components, utils);
-  var updatedRuleInstructions = util._extend({}, ruleInstructions);
-  Object.keys(detectionInstructions).forEach(function(instruction) {
-    updatedRuleInstructions[instruction] = function(node) {
+  const ruleInstructions = rule(context, components, utils);
+  const updatedRuleInstructions = util._extend({}, ruleInstructions);
+  Object.keys(detectionInstructions).forEach((instruction) => {
+    updatedRuleInstructions[instruction] = (node) => {
       detectionInstructions[instruction](node);
       return ruleInstructions[instruction] ? ruleInstructions[instruction](node) : void 0;
     };
@@ -398,7 +406,7 @@ function componentRule(rule, context) {
   return updatedRuleInstructions;
 }
 
-Components.detect = function(rule) {
+Components.detect = function (rule) {
   return componentRule.bind(this, rule);
 };
 

--- a/lib/util/stylesheet.js
+++ b/lib/util/stylesheet.js
@@ -118,7 +118,7 @@ const astHelpers = {
       );
     }
 
-    return [astHelpers.getStyleReferenceFromExpression(node.expression)];
+    return astHelpers.getStyleReferenceFromNode(node.expression);
   },
 
   collectStyleReferencesFromContainers: function (nodes) {

--- a/lib/util/stylesheet.js
+++ b/lib/util/stylesheet.js
@@ -1,0 +1,49 @@
+
+/**
+ * StyleSheets represents the StyleSheets found in the source code.
+ * @constructor
+ */
+function StyleSheets() {
+  this._styleSheets = {};
+}
+
+/**
+ * Add adds a StyleSheet to our StyleSheets collections.
+ *
+ * @param {string} styleSheetName - The name of the StyleSheet.
+ * @param {object} properties - The collection of rules in the styleSheet.
+ */
+StyleSheets.prototype.add = function(styleSheetName, properties) {
+  this._styleSheets[styleSheetName] = properties;
+}
+
+/**
+ * MarkAsUsed marks a rule as used in our source code by removing it from the
+ * specified StyleSheet rules.
+ *
+ * @param {string} fullyQualifiedName - The fully qualified name of the rule.
+ * for example 'styles.text'
+ */
+StyleSheets.prototype.markAsUsed = function(fullyQualifiedName) {
+  var nameSplit = fullyQualifiedName.split('.');
+  var styleSheetName = nameSplit[0];
+  var styleSheetProperty = nameSplit[1];
+
+  if (this._styleSheets[styleSheetName]) {
+    this._styleSheets[styleSheetName] = this
+      ._styleSheets[styleSheetName]
+      .filter(function(property) {
+        return property.key.name !== styleSheetProperty;
+      });
+  }
+}
+
+/**
+* GetUnusedReferences returns all collected StyleSheets and their
+* unmarked rules.
+*/
+StyleSheets.prototype.getUnusedReferences = function() {
+  return this._styleSheets;
+}
+
+module.exports.StyleSheets = StyleSheets;

--- a/lib/util/stylesheet.js
+++ b/lib/util/stylesheet.js
@@ -1,4 +1,4 @@
-
+'use strict';
 /**
  * StyleSheets represents the StyleSheets found in the source code.
  * @constructor
@@ -13,9 +13,9 @@ function StyleSheets() {
  * @param {string} styleSheetName - The name of the StyleSheet.
  * @param {object} properties - The collection of rules in the styleSheet.
  */
-StyleSheets.prototype.add = function(styleSheetName, properties) {
+StyleSheets.prototype.add = function (styleSheetName, properties) {
   this._styleSheets[styleSheetName] = properties;
-}
+};
 
 /**
  * MarkAsUsed marks a rule as used in our source code by removing it from the
@@ -24,26 +24,177 @@ StyleSheets.prototype.add = function(styleSheetName, properties) {
  * @param {string} fullyQualifiedName - The fully qualified name of the rule.
  * for example 'styles.text'
  */
-StyleSheets.prototype.markAsUsed = function(fullyQualifiedName) {
-  var nameSplit = fullyQualifiedName.split('.');
-  var styleSheetName = nameSplit[0];
-  var styleSheetProperty = nameSplit[1];
+StyleSheets.prototype.markAsUsed = function (fullyQualifiedName) {
+  const nameSplit = fullyQualifiedName.split('.');
+  const styleSheetName = nameSplit[0];
+  const styleSheetProperty = nameSplit[1];
 
   if (this._styleSheets[styleSheetName]) {
     this._styleSheets[styleSheetName] = this
       ._styleSheets[styleSheetName]
-      .filter(function(property) {
-        return property.key.name !== styleSheetProperty;
-      });
+      .filter((property) => property.key.name !== styleSheetProperty);
   }
-}
+};
 
 /**
 * GetUnusedReferences returns all collected StyleSheets and their
 * unmarked rules.
 */
-StyleSheets.prototype.getUnusedReferences = function() {
+StyleSheets.prototype.getUnusedReferences = function () {
   return this._styleSheets;
-}
+};
 
+const astHelpers = {
+  containsStyleSheetObject: function (node) {
+    return Boolean(
+      node &&
+      node.init &&
+      node.init.callee &&
+      node.init.callee.object &&
+      node.init.callee.object.name === 'StyleSheet'
+    );
+  },
+
+  containsCreateCall: function (node) {
+    return Boolean(
+      node &&
+      node.init &&
+      node.init.callee &&
+      node.init.callee.property &&
+      node.init.callee.property.name === 'create'
+    );
+  },
+
+  isStyleSheetDeclaration: function (node) {
+    return Boolean(
+      astHelpers.containsStyleSheetObject(node) &&
+      astHelpers.containsCreateCall(node)
+    );
+  },
+
+  getStyleSheetName: function (node) {
+    if (
+      node &&
+      node.id
+    ) {
+      return node.id.name;
+    }
+  },
+
+  getStyleDeclarations: function (node) {
+    if (
+      node &&
+      node.init &&
+      node.init.arguments &&
+      node.init.arguments[0] &&
+      node.init.arguments[0].properties
+    ) {
+      return node
+        .init
+        .arguments[0]
+        .properties;
+    }
+
+    return [];
+  },
+
+  isStyleAttribute: function (node) {
+    return Boolean(
+      node.type === 'JSXAttribute' &&
+      node.name &&
+      node.name.name &&
+      node.name.name.toLowerCase().includes('style')
+    );
+  },
+
+  collectStyleReferences: function (node) {
+    if (astHelpers.hasArrayOfStyleReferences(node)) {
+      const styleReferenceContainers = node
+        .expression
+        .elements;
+
+      return astHelpers.collectStyleReferencesFromContainers(
+        styleReferenceContainers
+      );
+    }
+
+    return [astHelpers.getStyleReferenceFromExpression(node.expression)];
+  },
+
+  collectStyleReferencesFromContainers: function (nodes) {
+    let styleReferences = [];
+    nodes.forEach((node) => {
+      styleReferences = styleReferences.concat(astHelpers.getStyleReferenceFromNode(node));
+    });
+
+    return styleReferences;
+  },
+
+  getStyleReferenceFromNode: function (node) {
+    let styleReference;
+    let leftStyleReferences;
+    let rightStyleReferences;
+
+    switch (node.type) {
+      case 'MemberExpression':
+        styleReference = astHelpers.getStyleReferenceFromExpression(node);
+        return [styleReference];
+      case 'LogicalExpression':
+        leftStyleReferences = astHelpers.getStyleReferenceFromNode(node.left);
+        rightStyleReferences = astHelpers.getStyleReferenceFromNode(node.right);
+        return [].concat(leftStyleReferences).concat(rightStyleReferences);
+      case 'ConditionalExpression':
+        leftStyleReferences = astHelpers.getStyleReferenceFromNode(node.consequent);
+        rightStyleReferences = astHelpers.getStyleReferenceFromNode(node.alternate);
+        return [].concat(leftStyleReferences).concat(rightStyleReferences);
+      default:
+        return [];
+    }
+  },
+
+  hasArrayOfStyleReferences: function (node) {
+    return Boolean(
+      node.type === 'JSXExpressionContainer' &&
+      node.expression &&
+      node.expression.type === 'ArrayExpression'
+    );
+  },
+
+  getStyleReferenceFromExpression: function (node) {
+    const result = [];
+    const name = astHelpers.getObjectName(node);
+    if (name) {
+      result.push(name);
+    }
+
+    const property = astHelpers.getPropertyName(node);
+    if (property) {
+      result.push(property);
+    }
+
+    return result.join('.');
+  },
+
+  getObjectName: function (node) {
+    if (
+      node &&
+      node.object &&
+      node.object.name
+    ) {
+      return node.object.name;
+    }
+  },
+
+  getPropertyName: function (node) {
+    if (
+      node &&
+      node.property &&
+      node.property.name
+    ) {
+      return node.property.name;
+    }
+  },
+};
+
+module.exports.astHelpers = astHelpers;
 module.exports.StyleSheets = StyleSheets;

--- a/lib/util/variable.js
+++ b/lib/util/variable.js
@@ -11,11 +11,11 @@
  * @returns {Boolean} True if the variable was found and marked as used, false if not.
  */
 function markVariableAsUsed(context, name) {
-  var scope = context.getScope();
-  var variables;
-  var i;
-  var len;
-  var found = false;
+  let scope = context.getScope();
+  let variables;
+  let i;
+  let len;
+  let found = false;
 
   // Special Node.js scope means we need to start one level deeper
   if (scope.type === 'global') {
@@ -45,8 +45,8 @@ function markVariableAsUsed(context, name) {
  * @returns {Boolean} True if the variable was found, false if not.
  */
 function findVariable(variables, name) {
-  var i;
-  var len;
+  let i;
+  let len;
 
   for (i = 0, len = variables.length; i < len; i++) {
     if (variables[i].name === name) {
@@ -67,8 +67,8 @@ function findVariable(variables, name) {
  * @returns {Boolean} True if the variable was found, false if not.
  */
 function variablesInScope(context) {
-  var scope = context.getScope();
-  var variables = scope.variables;
+  let scope = context.getScope();
+  let variables = scope.variables;
 
   while (scope.type !== 'global') {
     scope = scope.upper;
@@ -87,5 +87,5 @@ function variablesInScope(context) {
 module.exports = {
   findVariable: findVariable,
   variablesInScope: variablesInScope,
-  markVariableAsUsed: markVariableAsUsed
+  markVariableAsUsed: markVariableAsUsed,
 };

--- a/package.json
+++ b/package.json
@@ -6,7 +6,7 @@
   "main": "index.js",
   "scripts": {
     "coveralls": "cat ./reports/coverage/lcov.info | coveralls",
-    "lint": "eslint ./lib",
+    "lint": "eslint ./lib && eslint ./tests",
     "test": "npm run lint && npm run unit-test",
     "unit-test": "istanbul cover --dir reports/coverage node_modules/mocha/bin/_mocha tests/**/*.js -- --reporter dot"
   },

--- a/package.json
+++ b/package.json
@@ -6,7 +6,7 @@
   "main": "index.js",
   "scripts": {
     "coveralls": "cat ./reports/coverage/lcov.info | coveralls",
-    "lint": "eslint ./",
+    "lint": "eslint ./lib",
     "test": "npm run lint && npm run unit-test",
     "unit-test": "istanbul cover --dir reports/coverage node_modules/mocha/bin/_mocha tests/**/*.js -- --reporter dot"
   },
@@ -23,9 +23,11 @@
   "homepage": "https://github.com/intellicode/eslint-plugin-react-native",
   "bugs": "https://github.com/intellicode/eslint-plugin-react-native/issues",
   "devDependencies": {
-    "babel-eslint": "4.1.8",
+    "babel-eslint": "6.0.0",
     "coveralls": "2.11.4",
-    "eslint": "2.0.0",
+    "eslint": "2.4.0",
+    "eslint-config-airbnb": "6.2.0",
+    "eslint-plugin-react": "4.2.3",
     "istanbul": "0.4.0",
     "mocha": "2.3.3"
   },

--- a/package.json
+++ b/package.json
@@ -25,7 +25,7 @@
   "devDependencies": {
     "babel-eslint": "6.0.0",
     "coveralls": "2.11.4",
-    "eslint": "2.4.0",
+    "eslint": "2.5.1",
     "eslint-config-airbnb": "6.2.0",
     "eslint-plugin-react": "4.2.3",
     "istanbul": "0.4.0",

--- a/tests/index.js
+++ b/tests/index.js
@@ -1,24 +1,22 @@
 /* eslint-env mocha */
 'use strict';
 
-var plugin = require('..');
+const plugin = require('..');
 
-var assert = require('assert');
-var fs = require('fs');
-var path = require('path');
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
 
-var rules = fs.readdirSync(path.resolve(__dirname, '../lib/rules/'))
-  .map(function(f) {
-    return path.basename(f, '.js');
-  });
+const rules = fs.readdirSync(path.resolve(__dirname, '../lib/rules/'))
+  .map((f) => path.basename(f, '.js'));
 
-var defaultSettings = {
-  'jsx-uses-vars': 1
+const defaultSettings = {
+  'jsx-uses-vars': 1,
 };
 
-describe('all rule files should be exported by the plugin', function() {
-  rules.forEach(function(ruleName) {
-    it('should export ' + ruleName, function() {
+describe('all rule files should be exported by the plugin', () => {
+  rules.forEach((ruleName) => {
+    it('should export ' + ruleName, () => {
       assert.equal(
         plugin.rules[ruleName],
         require(path.join('../lib/rules', ruleName))
@@ -26,15 +24,15 @@ describe('all rule files should be exported by the plugin', function() {
     });
 
     if (defaultSettings.hasOwnProperty(ruleName)) {
-      var val = defaultSettings[ruleName];
-      it('should configure ' + ruleName + ' to ' + val + ' by default', function() {
+      const val = defaultSettings[ruleName];
+      it('should configure ' + ruleName + ' to ' + val + ' by default', () => {
         assert.equal(
           plugin.rulesConfig[ruleName],
           val
         );
       });
     } else {
-      it('should configure ' + ruleName + ' off by default', function() {
+      it('should configure ' + ruleName + ' off by default', () => {
         assert.equal(
           plugin.rulesConfig[ruleName],
           0

--- a/tests/lib/rules/no-unused-styles.js
+++ b/tests/lib/rules/no-unused-styles.js
@@ -8,8 +8,8 @@
 // Requirements
 // ------------------------------------------------------------------------------
 
-var rule = require('../../../lib/rules/no-unused-styles');
-var RuleTester = require('eslint').RuleTester;
+const rule = require('../../../lib/rules/no-unused-styles');
+const RuleTester = require('eslint').RuleTester;
 
 require('babel-eslint');
 
@@ -17,7 +17,7 @@ require('babel-eslint');
 // Tests
 // ------------------------------------------------------------------------------
 
-var ruleTester = new RuleTester();
+const ruleTester = new RuleTester();
 ruleTester.run('no-unused-styles', rule, {
 
   valid: [{
@@ -29,13 +29,13 @@ ruleTester.run('no-unused-styles', rule, {
       '  render: function() {',
       '    return <Text textStyle={styles.name}>Hello {this.props.name}</Text>;',
       '  }',
-      '});'
+      '});',
     ].join('\n'),
     parser: 'babel-eslint',
     ecmaFeatures: {
       classes: true,
-      jsx: true
-    }
+      jsx: true,
+    },
   }, {
     code: [
       'const styles = StyleSheet.create({',
@@ -45,13 +45,13 @@ ruleTester.run('no-unused-styles', rule, {
       '  render: function() {',
       '    return <Text style={styles.name}>Hello {this.props.name}</Text>;',
       '  }',
-      '});'
+      '});',
     ].join('\n'),
     parser: 'babel-eslint',
     ecmaFeatures: {
       classes: true,
-      jsx: true
-    }
+      jsx: true,
+    },
   }, {
     code: [
       'const styles = StyleSheet.create({',
@@ -67,13 +67,13 @@ ruleTester.run('no-unused-styles', rule, {
       '  render: function() {',
       '    return <Text style={styles.welcome}>Welcome</Text>;',
       '  }',
-      '});'
+      '});',
     ].join('\n'),
     parser: 'babel-eslint',
     ecmaFeatures: {
       classes: true,
-      jsx: true
-    }
+      jsx: true,
+    },
   }, {
     code: [
       'const styles = StyleSheet.create({',
@@ -86,13 +86,13 @@ ruleTester.run('no-unused-styles', rule, {
       '  render: function() {',
       '    return <Text style={[styles.text, textStyle]}>Hello {this.props.name}</Text>;',
       '  }',
-      '});'
+      '});',
     ].join('\n'),
     parser: 'babel-eslint',
     ecmaFeatures: {
       classes: true,
-      jsx: true
-    }
+      jsx: true,
+    },
   }, {
     code: [
       'const styles = StyleSheet.create({',
@@ -106,15 +106,19 @@ ruleTester.run('no-unused-styles', rule, {
       '    textStyle: Text.propTypes.style,',
       '  },',
       '  render: function() {',
-      '    return <Text style={[styles.text, styles2.text, textStyle]}>Hello {this.props.name}</Text>;',
+      '    return (',
+      '      <Text style={[styles.text, styles2.text, textStyle]}>',
+      '       Hello {this.props.name}',
+      '      </Text>',
+      '     );',
       '  }',
-      '});'
+      '});',
     ].join('\n'),
     parser: 'babel-eslint',
     ecmaFeatures: {
       classes: true,
-      jsx: true
-    }
+      jsx: true,
+    },
   }, {
     code: [
       'const styles = StyleSheet.create({',
@@ -135,13 +139,13 @@ ruleTester.run('no-unused-styles', rule, {
       '      </Text>',
       '    );',
       '  }',
-      '});'
+      '});',
     ].join('\n'),
     parser: 'babel-eslint',
     ecmaFeatures: {
       classes: true,
-      jsx: true
-    }
+      jsx: true,
+    },
   }, {
     code: [
       'const styles = StyleSheet.create({',
@@ -153,15 +157,19 @@ ruleTester.run('no-unused-styles', rule, {
       '    return { condition: true }; ',
       '  },',
       '  render: function() {',
-      '    return <Text style={[this.state.condition ? styles.text : styles.text2]}>Hello {this.props.name}</Text>;',
+      '    return (',
+      '      <Text style={[this.state.condition ? styles.text : styles.text2]}>',
+      '        Hello {this.props.name}',
+      '      </Text>',
+      '    );',
       '  }',
-      '});'
+      '});',
     ].join('\n'),
     parser: 'babel-eslint',
     ecmaFeatures: {
       classes: true,
-      jsx: true
-    }
+      jsx: true,
+    },
   }, {
     code: [
       'const styles = StyleSheet.create({',
@@ -171,8 +179,8 @@ ruleTester.run('no-unused-styles', rule, {
     parser: 'babel-eslint',
     ecmaFeatures: {
       classes: true,
-      jsx: true
-    }
+      jsx: true,
+    },
   }],
 
   invalid: [{
@@ -184,15 +192,15 @@ ruleTester.run('no-unused-styles', rule, {
       '  render: function() {',
       '    return <Text style={styles.b}>Hello {this.props.name}</Text>;',
       '  }',
-      '});'
+      '});',
     ].join('\n'),
     parser: 'babel-eslint',
     ecmaFeatures: {
       classes: true,
-      jsx: true
+      jsx: true,
     },
     errors: [{
-      message: 'Unused style detected: styles.text'
-    }]
-  }
-]});
+      message: 'Unused style detected: styles.text',
+    }],
+  },
+] });

--- a/tests/lib/rules/no-unused-styles.js
+++ b/tests/lib/rules/no-unused-styles.js
@@ -173,6 +173,30 @@ ruleTester.run('no-unused-styles', rule, {
   }, {
     code: [
       'const styles = StyleSheet.create({',
+      '    style1: {',
+      '        color: \'red\',',
+      '    },',
+      '    style2: {',
+      '        color: \'blue\',',
+      '    }',
+      '});',
+      'export default class MyComponent extends Component {',
+      '    static propTypes = {',
+      '        isDanger: PropTypes.bool',
+      '    };',
+      '    render() {',
+      '        return <View style={this.props.isDanger ? styles.style1 : styles.style2} />;',
+      '    }',
+      '}',
+    ].join('\n'),
+    parser: 'babel-eslint',
+    ecmaFeatures: {
+      classes: true,
+      jsx: true,
+    },
+  }, {
+    code: [
+      'const styles = StyleSheet.create({',
       '  text: {}',
       '})',
     ].join('\n'),

--- a/tests/lib/rules/split-platform-components.js
+++ b/tests/lib/rules/split-platform-components.js
@@ -9,8 +9,8 @@
 // Requirements
 // ------------------------------------------------------------------------------
 
-var rule = require('../../../lib/rules/split-platform-components');
-var RuleTester = require('eslint').RuleTester;
+const rule = require('../../../lib/rules/split-platform-components');
+const RuleTester = require('eslint').RuleTester;
 
 require('babel-eslint');
 
@@ -18,111 +18,111 @@ require('babel-eslint');
 // Tests
 // ------------------------------------------------------------------------------
 
-var ruleTester = new RuleTester();
+const ruleTester = new RuleTester();
 ruleTester.run('no-unused-styles', rule, {
 
   valid: [{
     code: [
-      'var React = require(\'react-native\');',
-      'var {',
+      'const React = require(\'react-native\');',
+      'const {',
       '  ActivityIndicatiorIOS,',
       '} = React',
       'const Hello = React.createClass({',
       '  render: function() {',
       '    return <ActivityIndicatiorIOS />;',
       '  }',
-      '});'
+      '});',
     ].join('\n'),
     filename: 'Hello.ios.js',
     parser: 'babel-eslint',
     ecmaFeatures: {
       classes: true,
-      jsx: true
-    }
+      jsx: true,
+    },
   }, {
     code: [
-      'var React = require(\'react-native\');',
-      'var {',
+      'const React = require(\'react-native\');',
+      'const {',
       '  ProgressBarAndroid,',
       '} = React',
       'const Hello = React.createClass({',
       '  render: function() {',
       '    return <ProgressBarAndroid />;',
       '  }',
-      '});'
+      '});',
     ].join('\n'),
     parser: 'babel-eslint',
     filename: 'Hello.android.js',
     ecmaFeatures: {
       classes: true,
-      jsx: true
-    }
+      jsx: true,
+    },
   }, {
     code: [
-      'var React = require(\'react-native\');',
-      'var {',
+      'const React = require(\'react-native\');',
+      'const {',
       '  View,',
       '} = React',
       'const Hello = React.createClass({',
       '  render: function() {',
       '    return <View />;',
       '  }',
-      '});'
+      '});',
     ].join('\n'),
     parser: 'babel-eslint',
     filename: 'Hello.js',
     ecmaFeatures: {
       classes: true,
-      jsx: true
-    }
+      jsx: true,
+    },
   }],
 
   invalid: [{
     code: [
-      'var React = require(\'react-native\');',
-      'var {',
+      'const React = require(\'react-native\');',
+      'const {',
       '  ProgressBarAndroid,',
       '} = React',
       'const Hello = React.createClass({',
       '  render: function() {',
       '    return <ProgressBarAndroid />;',
       '  }',
-      '});'
+      '});',
     ].join('\n'),
     parser: 'babel-eslint',
     filename: 'Hello.js',
     ecmaFeatures: {
       classes: true,
-      jsx: true
+      jsx: true,
     },
     errors: [{
-      message: 'Android components should be placed in android files'
-    }]
+      message: 'Android components should be placed in android files',
+    }],
   }, {
     code: [
-      'var React = require(\'react-native\');',
-      'var {',
+      'const React = require(\'react-native\');',
+      'const {',
       '  ActivityIndicatiorIOS,',
       '} = React',
       'const Hello = React.createClass({',
       '  render: function() {',
       '    return <ActivityIndicatiorIOS />;',
       '  }',
-      '});'
+      '});',
     ].join('\n'),
     parser: 'babel-eslint',
     filename: 'Hello.js',
     ecmaFeatures: {
       classes: true,
-      jsx: true
+      jsx: true,
     },
     errors: [{
-      message: 'IOS components should be placed in ios files'
-    }]
+      message: 'IOS components should be placed in ios files',
+    }],
   }, {
     code: [
-      'var React = require(\'react-native\');',
-      'var {',
+      'const React = require(\'react-native\');',
+      'const {',
       '  ActivityIndicatiorIOS,',
       '  ProgressBarAndroid,',
       '} = React',
@@ -130,18 +130,18 @@ ruleTester.run('no-unused-styles', rule, {
       '  render: function() {',
       '    return <ActivityIndicatiorIOS />;',
       '  }',
-      '});'
+      '});',
     ].join('\n'),
     parser: 'babel-eslint',
     filename: 'Hello.js',
     ecmaFeatures: {
       classes: true,
-      jsx: true
+      jsx: true,
     },
     errors: [{
-      message: 'IOS and Android components can\'t be mixed'
+      message: 'IOS and Android components can\'t be mixed',
     }, {
-      message: 'IOS and Android components can\'t be mixed'
-    }]
-  }
-]});
+      message: 'IOS and Android components can\'t be mixed',
+    }],
+  },
+] });

